osu.Game.Rulesets.Catch/Edit/CatchSelectionHandler.cs

```diff
@@ -26,13 +29,19 @@ public override bool HandleMovement(MoveSelectionEvent<HitObject> moveEvent)
             Vector2 targetPosition = HitObjectContainer.ToLocalSpace(blueprint.ScreenSpaceSelectionPoint + moveEvent.ScreenSpaceDelta);
             float deltaX = targetPosition.X - originalPosition.X;
 
+            foreach (float x in EditorBeatmap.SelectedHitObjects.SelectMany(getOriginalPositions))
+                deltaX = Math.Clamp(deltaX, -x, CatchPlayfield.WIDTH - x);
```

Comment:
I don't know if it's just me but I had to stop and think for 10 minutes about what this is doing. It looks to be correct, because from what I can tell, what it's doing is that it's computing the allowable range of movement for each `x` coordinate returned by `getOriginalPositions` (which starts at `-x` - `x` units to the left, with the sign flipped, and ends at `WIDTH - x` units to the right), and then doing an intersection of those ranges via the iterated clamps, but I'm not super sure that should be presented just like this without comment.